dicodile/workers/dicod_worker.py

```diff
@@ -668,7 +669,12 @@ def recv_D(self):
         """Receive a dictionary D"""
         if self._backend == "mpi":
             comm = MPI.Comm.Get_parent()
-            self.D = recv_broadcasted_array(comm)
+            if self.rank1:
+                self.u = recv_broadcasted_array(comm)
+                self.v = recv_broadcasted_array(comm)
+                self.D = get_D(self.u, self.v)
```

Comment:
The purpose here is to keep `self.u, self.v` instead of constructing the full one.
I think you don't need it anymore now and the following should work:
```suggestion
                self.D = (self.u, self.v)
```

Reply:
Yes, although doing that made me realize that we extract a variant of `n_channels, n_atoms, *atom_support` from D.shape in numerous places in that codes, so encapsulating that would be nice -- I'll get on it.